File: app/src/main/java/com/kode/utair/data/repository/WeatherRepository.java
package com.kode.utair.data.repository;

import com.kode.utair.data.repository.datasource.CloudWeatherDataSource;
import com.kode.utair.domain.models.WeatherForecastEntity;
import com.kode.utair.domain.repository.IWeatherRepository;

import javax.inject.Inject;

import io.reactivex.Single;

public class WeatherRepository implements IWeatherRepository {

    private CloudWeatherDataSource cloudWeatherDataSource;

    @Inject
    public WeatherRepository(CloudWeatherDataSource cloudWeatherDataSource) {
        this.cloudWeatherDataSource = cloudWeatherDataSource;
    }

    @Override
    public Single<WeatherForecastEntity> getWeatherForecastForCity(String cityName) {
        return cloudWeatherDataSource.getWeather(cityName);
    }



}

File: app/src/main/java/com/kode/utair/data/repository/WeatherRepository.kt
package com.kode.utair.data.repository

import com.kode.utair.data.network.WeatherApiService
import com.kode.utair.data.network.mappers.WeatherForecastResponseMapper
import com.kode.utair.domain.models.WeatherForecastEntity
import com.kode.utair.domain.repository.IWeatherRepository
import io.reactivex.Single
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import javax.inject.Inject

class WeatherRepository @Inject constructor(
        private val weatherApiService: WeatherApiService,
        private val weatherForecastResponseMapper: WeatherForecastResponseMapper
) : IWeatherRepository {

    override fun getWeatherForecastForCity(cityName: String): Single<WeatherForecastEntity> {
        return weatherApiService.getWeather(cityName)
                .map { weatherForecastResponseMapper.map(it) }
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())

    }

}